### PR TITLE
small patch for EDNS0 cookie support

### PR DESCRIPTION
**Pull request by shuque**
If you're interested, a small patch to add EDNS0 cookie support.

### CODE REVIEW

File: edns.go [hunk 1]
```diff
@@ -286,6 +289,32 @@ func (e *EDNS0_SUBNET) String() (s string) {
...
+//o := new(dns.OPT)
+//o.Hdr.Name = "."
+//o.Hdr.Rrtype = dns.TypeOPT
+//e := new(dns.EDNS0_COOKIE)
+//e.Code = dns.EDNS0COOKIE
+//e.Cookie = "24a5ac.."
+//o.Option = append(o.Option, e)
+type EDNS0_COOKIE struct {
+	Code uint16                   // Always EDNS0COOKIE
+	Cookie string                 // Hex-encoded cookie data
```
Comment (miekg) [inline edns.go]: Has this code been gofmt-ed?

File: edns.go [hunk 2]
```diff
@@ -286,6 +289,32 @@ func (e *EDNS0_SUBNET) String() (s string) {
...
+	h, err := hex.DecodeString(e.Cookie)
+	if err != nil {
+		return nil, err
+	}
+	return h, nil
+}
+
+func (e *EDNS0_COOKIE) Option() uint16        { return EDNS0COOKIE }
+func (e *EDNS0_COOKIE) unpack(b []byte) error { e.Cookie = hex.EncodeToString(b); return nil }
+func (e *EDNS0_COOKIE) String() string        { return string(e.Cookie) }
```
Comment (miekg) [inline edns.go]: This does not need to string conversion, e.Cookie is already a string

**Comment by miekg** [comment]
some nits and could update the README and add the RFC number (is that already known) to it?

File: edns.go (hunk 2 above)
Comment (shuque) [inline edns.go]: Yup, will fix.

**Comment by shuque** [comment]
Thanks. I'll fix the nits and update README. 
RFC not yet published, but is close and the official IANA EDNS option code is already allocated (10) - instead I can cite the latest draft:
https://tools.ietf.org/html/draft-ietf-dnsop-cookies-10

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [miekg/dns] small patch for EDN..." ]
>Thanks. I'll fix the nits and update README.
>RFC not yet published, but is close and the official IANA EDNS option code is already allocated (10) - instead I can cite the latest draft:
>https://tools.ietf.org/html/draft-ietf-dnsop-cookies-10

that will work as well, I just want all the drafts/RFCs implemented to be listed
there.

File: edns.go [hunk 3]
```diff
@@ -286,6 +289,32 @@ func (e *EDNS0_SUBNET) String() (s string) {
 	return
 }
 
+// The Cookie EDNS0 option
+//
+//o := new(dns.OPT)
```
Comment (miekg) [inline edns.go]: another nit: should be a space here as well

Comment (shuque) [inline edns.go]: Ah yes, looks like you fixed the spaces already. Let me know if you wanted me to do anything else.

**Comment by bortzmeyer** [comment]
Cookies are now  RFC 7873 so you may want to update README.md.

**Pull request merged**